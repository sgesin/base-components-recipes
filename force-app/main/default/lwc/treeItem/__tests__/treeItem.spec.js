/*
 * Copyright (c) 2019, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */

import { createElement } from 'lwc';
import { shadowQuerySelector } from 'lightning/testUtils';
import Element from 'c/treeItem';

describe('c-tree-item', () => {
    it('renders leaf tree-item', () => {
        const element = createElement('c-tree-item', { is: Element });
        document.body.appendChild(element);
        element.isRoot = false;
        element.label = 'label';
        element.href = 'href';
        element.metatext = 'metatext';
        element.nodeRef = {
            expanded: false
        };

        element.isExpanded = false;
        element.isDisabled = false;
        element.nodename = '2234';
        element.nodeKey = '1.1';
        element.isLeaf = true;

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('should render disabled tree item', () => {
        const element = createElement('c-tree-item', { is: Element });

        Object.assign(element, {
            isRoot: false,
            label: 'label',
            metatext: 'metatext',
            isExpanded: true,
            isDisabled: true,
            ariaDisabled: true
        });

        document.body.appendChild(element);

        return Promise.resolve().then(() => {
            const itemEl = element.shadowRoot.querySelector('.slds-tree__item');
            const buttonEl = itemEl.querySelector('button');
            const labelEl = itemEl.querySelector('.slds-tree__item-label');
            const metaEl = itemEl.querySelector('.slds-tree__item-meta');

            expect(itemEl.querySelector('a')).toBe(null);
            expect(buttonEl.classList.contains('slds-hidden')).toBe(true);
            expect(labelEl.textContent).toBe('label');
            expect(metaEl.textContent).toBe(':metatext');
        });
    });

    it('should hide children items of disabled item', () => {
        const element = createElement('c-tree-item', { is: Element });

        Object.assign(element, {
            isRoot: false,
            label: 'label',
            metatext: 'metatext',
            isExpanded: true,
            isDisabled: true,
            ariaDisabled: true,
            nodeRef: {
                expanded: true
            },

            childItems: [
                {
                    isRoot: false,
                    label: 'label-child',
                    isExpanded: false,
                    isDisabled: false,
                    href: 'href',
                    isLeaf: true,
                    key: 'key'
                }
            ]
        });

        document.body.appendChild(element);

        return Promise.resolve().then(() => {
            expect(element.shadowRoot.querySelector('c-tree-item')).toBe(null);
        });
    });

    it('uses the correct icon for LTR', () => {
        const element = createElement('c-tree-item', { is: Element });
        document.body.appendChild(element);

        element.isRoot = false;
        element.label = 'label';
        element.href = 'href';

        return Promise.resolve().then(() => {
            const lightningIcon = shadowQuerySelector(
                element,
                'c-primitive-icon'
            );

            expect(lightningIcon.iconName).toBe('utility:chevronright');
        });
    });

    it('uses the correct icon for RTL', () => {
        const element = createElement('c-tree-item', { is: Element });
        document.dir = 'rtl';
        document.body.appendChild(element);

        element.isRoot = false;
        element.label = 'label';
        element.href = 'href';

        return Promise.resolve().then(() => {
            const lightningIcon = shadowQuerySelector(
                element,
                'c-primitive-icon'
            );

            expect(lightningIcon.iconName).toBe('utility:chevronleft');
        });
    });
});
