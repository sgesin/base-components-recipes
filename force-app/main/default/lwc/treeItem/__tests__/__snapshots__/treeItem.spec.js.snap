// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-tree-item renders leaf tree-item 1`] = `
<c-tree-item>
  #shadow-root(open)
    <div
      class="slds-tree__item"
    >
      <button
        aria-hidden="true"
        class="slds-button slds-button_icon slds-m-right_x-small slds-hidden"
        tabindex="-1"
        title="Expand Tree Branch"
        type="button"
      >
        <c-primitive-icon>
          #shadow-root(open)
            <svg
              aria-hidden="true"
              class="slds-button__icon slds-button__icon_small"
              data-key="chevronright"
              focusable="false"
            >
              <use
                xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
              />
            </svg>
        </c-primitive-icon>
        <span
          class="slds-assistive-text"
        >
          Expand Tree Branch
        </span>
      </button>
      <span
        class="slds-has-flexi-truncate"
      >
        <a
          href="href"
          role="presentation"
          tabindex="-1"
        >
          <span
            class="slds-tree__item-label slds-truncate"
            title="label"
          >
            label
          </span>
          <span
            class="slds-tree__item-meta slds-truncate"
            title="metatext"
          >
            <span
              class="slds-assistive-text"
            >
              :
            </span>
            metatext
          </span>
        </a>
      </span>
    </div>
</c-tree-item>
`;
